draft-ietf-webpush-encryption.md

```diff
@@ -108,170 +101,277 @@ encryption scheme by applications that use push messaging.  An agent that only
 delivers messages that are properly encrypted strongly encourages the end-to-end
 protection of messages.
 
-A web browser that implements the Web Push API [API] can enforce the use of
+A web browser that implements the Web Push API {{API}} can enforce the use of
 encryption by forwarding only those messages that were properly encrypted.
 
 
 ## Notational Conventions
 
 The words "MUST", "MUST NOT", "SHOULD", and "MAY" are used in this document.
 It's not shouting, when they are capitalized, they have the special meaning
-described in [RFC2119].
+described in {{!RFC2119}}.
 
 
-# Key Generation and Agreement
+# Push Message Encryption Overview {#overview}
+
+Encrypting a push message uses elliptic-curve Diffie-Hellman (ECDH) {{ECDH}} to
+establish a shared secret (see {{dh}}) and a symmetric secret for authentication
+(see {{auth}}).
+
+A User Agent generates an ECDH key pair and authentication secret that it
+associates with each subscription it creates.  These are sent to the Application
+Server with other details of the push subscription.
+
+When sending a message, an Application Server generates an ECDH key pair and a
+random salt.  The ECDH public key is encoded into the `dh` parameter of the
+Crypto-Key header field; the salt is encoded into the `salt` parameter of the
+Encryption header field.
+
```

Comment:
Actually, the key pair can be discarded after sending the message.  That's worth point out for sure.  (Yeah, smart people will try to save themselves a tiny amount of computation by reusing keys, we won't tell them not to, but the text can imply that discard is the right thing to do - because it is :)